### DISCUSSION TITLE
Field inversion for NACA airfoil not working in parallel.

### DISCUSSION THREAD

**Discussion by yannwadham** [mdolab/dafoam#495] [General]
Hi, 

I am very new to DAFoam and I am trying to run a field inversion on a 2D NACA airfoil. 
I adapted the runscript.py file from the periodic hill tutorial and while it works well in serial I don't manage to run it in parallel.
Here is the error message I obtain when I run the same script in parallel using : mpirun -np 8 python runScript_vYY.py 2>&1 | tee logOpt.txt
4032 is my cell count 
![Error_message](https://github.com/mdolab/dafoam/assets/128370345/8a92898b-af5b-4aa7-8588-15fdbe4b5e03)

Here is the case attached
[case.zip](https://github.com/mdolab/dafoam/files/12804851/case.zip)

Do you know what am I doing wrong? 

Thank you very much in advance.

**Comment by friedenhe** [comment]
Thanks for the feedback. We reproduced the issue you saw. This is caused by a bug in the v3 version of runScript.py. So, please use runScript_v2.py for field inversion in parallel..

**Reply by yannwadham** [reply]
Ok! Thank you for the feedback, I will use the other script.

**Comment by lnhang0618** [comment]
Has this issue been fixed?

**Reply by lnhang0618** [reply]
runScripts_v2 is encountering convergence issues. I only changed the number of grids and the corresponding nCells in the script file. Despite not altering other settings, it is unusual that FI_beta_fieldInversion in runScripts_v2 is not equal to zero during runprimal.

**Comment by friedenhe** [comment]
Unfortunately, the field inversion works for v2 scripts only
